tests/suites/main_test.function

```diff
@@ -395,7 +405,50 @@ int main(int argc, const char *argv[])
             if( unmet_dep_count == 0 )
             {
                 test_errors = 0;
+
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+                /* Suppress all output from the library unless we're verbose
+                 * mode
+                 */
+                if( !option_verbose )
+                {
+                    /* Redirect all stdout output to /dev/null */
+                    tests_stdout = dup( fileno(stdout) );
+                    if( tests_stdout == -1 )
+                    {
+                        /* Redirection has failed with no stdout so exit */
+                        exit(1);
+                    }
+
+                    fflush( stdout );
+                    fclose( stdout );
+                    stdout = fopen("/dev/null", "w" );
+                    if( stdout == NULL )
+                    {
+                        /* Redirection has failed with no stdout so exit */
+                        exit(1);
+                    }
+                }
+#endif /* __unix__ || __APPLE__ __MACH__ */
+
                 ret = dispatch_test( cnt, params );
+
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+                if( !option_verbose )
+                {
+                    /* Restore stdout */
+                    fflush( stdout );
+                    fclose( stdout );
+
+                    stdout = fdopen ( tests_stdout, "w");
```

Comment:
Style: space between function name and `(`. Also missing space between `"` and `)`.